From: Alexander Larsson <alexl@redhat.com>
Date: Tue, 17 Jan 2017 15:58:28 +0100
Subject: Call setsid() and setexeccon() befor forking the init monitor

This means such options get inherited by pid 1 too.

Closes: #156
Approved by: cgwalters

Origin: upstream, 0.1.7, commit:c93370a4a5bc44a10cd77bafb6d2a53d27554064
---
 bubblewrap.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/bubblewrap.c b/bubblewrap.c
index 61f355f..0e4ec60 100644
--- a/bubblewrap.c
+++ b/bubblewrap.c
@@ -2080,6 +2080,13 @@ main (int    argc,
   xsetenv ("PWD", new_cwd, 1);
   free (old_cwd);
 
+  if (opt_new_session &&
+      setsid () == (pid_t) -1)
+    die_with_error ("setsid");
+
+  if (label_exec (opt_exec_label) == -1)
+    die_with_error ("label_exec %s", argv[0]);
+
   __debug__ (("forking for child\n"));
 
   if (opt_unshare_pid || lock_files != NULL || opt_sync_fd != -1)
@@ -2127,13 +2134,6 @@ main (int    argc,
   /* We want sigchild in the child */
   unblock_sigchild ();
 
-  if (opt_new_session &&
-      setsid () == (pid_t) -1)
-    die_with_error ("setsid");
-
-  if (label_exec (opt_exec_label) == -1)
-    die_with_error ("label_exec %s", argv[0]);
-
   if (execvp (argv[0], argv) == -1)
     die_with_error ("execvp %s", argv[0]);
 
